connectors/services/job_cleanup.py

```diff
@@ -0,0 +1,123 @@
+                await self._process_stuck_jobs()
+                await self._sleeps.sleep(self.idling)
+        finally:
+            if self.connector_index is not None:
+                self.connector_index.stop_waiting()
+                await self.connector_index.close()
+            if self.sync_job_index is not None:
+                self.sync_job_index.stop_waiting()
+                await self.sync_job_index.close()
+        return 0
+
+    async def _process_orphaned_jobs(self):
+        try:
+            logger.debug("Start cleaning up orphaned jobs...")
+            connector_ids = [
+                connector.id
+                async for connector in self.connector_index.all_connectors()
+            ]
+
+            content_indices = set()
+            job_ids = []
+            async for job in self.sync_job_index.orphaned_jobs(
+                connector_ids=connector_ids
+            ):
+                if job.index_name is not None:
+                    content_indices.add(job.index_name)
+                job_ids.append(job.job_id)
+
+            if len(job_ids) == 0:
+                logger.debug("No orphaned jobs found. Skipping...")
+                return
+
+            # delete content indices in case they are re-created by sync job
+            if len(content_indices) > 0:
+                await self.sync_job_index.delete_indices(indices=list(content_indices))
+            result = await self.sync_job_index.delete_jobs(job_ids=job_ids)
+            if len(result["failures"]) > 0:
+                logger.error(f"Error found when deleting jobs: {result['failures']}")
+            logger.info(
+                f"Successfully deleted {result['deleted']} out of {result['total']} orphaned jobs."
+            )
+        except Exception as e:
+            logger.critical(e, exc_info=True)
+            self.raise_if_spurious(e)
+
+    async def _process_stuck_jobs(self):
+        try:
+            logger.debug("Start cleaning up stuck jobs...")
+            connector_ids = [
+                connector.id
+                async for connector in self.connector_index.supported_connectors(
+                    native_service_types=self.native_service_types,
+                    connectors_ids=self.connectors_ids,
+                )
+            ]
+
+            marked_count = total_count = 0
+            async for job in self.sync_job_index.stuck_jobs(
+                connector_ids=connector_ids
+            ):
+                try:
+                    connector_id = job.connector_id
+                    connector = await self.connector_index.fetch_by_id(
+                        doc_id=connector_id
+                    )
+                    if connector is None:
+                        logger.warning(
+                            f"Could not found connector by id #{connector_id}"
+                        )
+                    else:
+                        await connector._sync_done(
```

Comment:
This line will give warnings in IDE as protected methods are not supposed to be accessed.

Inside the `_sync_done` method, this line will also raise error:

https://github.com/elastic/connectors-python/blob/020a512fd61443710bc777c5e04b9df09dc48443/connectors/byoc.py#L553-L556

Because `self._start_time` is never initialized. But the `sync_done` operation is actually done, it's just the log raising the error.

I'd like to leave it as this is going to be refactored in PR https://github.com/elastic/connectors-python/pull/375, and I don't want to introduce too much un-related changes in this PR